editor/modes/visual-editor/block.js

```diff
@@ -60,6 +61,44 @@ class VisualEditorBlock extends wp.element.Component {
 		}
 	}
 
+	mergeWithPrevious() {
+		const { block, previousBlock, onRemove, onChange } = this.props;
+
+		// Do nothing when it's the first block
+		if ( ! previousBlock ) {
+			return;
+		}
+
+		const previousBlockSettings = wp.blocks.getBlockSettings( previousBlock.blockType );
+
+		// Remove the previous block if it's not mergeable
+		if ( ! previousBlockSettings.merge ) {
+			onRemove( previousBlock.uid );
+			return;
+		}
+
+		// We can only merge blocks with similar types
+		// thus, we transform the block to merge first
+		const blockWithTheSameType = previousBlock.blockType === block.blockType
+			? block
+			: wp.blocks.switchToBlockType( block, previousBlock.blockType );
+
+		// If the block types can not match, do nothing
+		if ( ! blockWithTheSameType ) {
+			return;
```

Comment:
My main reason for bringing it up was merely as a point of consistency, since it seemed from the above condition that the "unhandleable" case should simply be removed.